File: java/chrishenx/math_expression_system/Symbol.java
package chrishenx.math_expression_system;

public final class Symbol {
	public final boolean isTerminal;
	public final TerminalSymbolType terminalSymbol;
	public final NonTerminalSymbol nonTerminalSymbol;
	
	public Symbol(TerminalSymbolType terminalSymbol) {
		this.isTerminal = true;
		this.terminalSymbol = terminalSymbol;
		this.nonTerminalSymbol = null;
	}
	
	public Symbol(NonTerminalSymbol nonTerminalSymbol) {
		this.isTerminal = false;
		this.nonTerminalSymbol = nonTerminalSymbol;
		this.terminalSymbol = null;
	}
}

File: c-sharp/library/Symbol.cs
//=======================================================================
//  Symbol.cs
//  math-expressions-system/c-sharp
//  Created by Christian González on 10/04/15.
//
//  Copyright © Christian González 2015.
//
//  Distributed under the MIT License.
//  (See accompanying file LICENSE or copy at
//  http://opensource.org/licenses/MIT)
//=======================================================================

namespace chrishenx.MathExpressionSystem
{
    sealed class Symbol
    {
        public readonly bool IsTerminal;
        public readonly TerminalSymbolType TerminalSymbol;
        public readonly NonTerminalSymbol NonTerminalSymbol;

        public Symbol(TerminalSymbolType terminalSymbol)
        {
            IsTerminal = true;
            TerminalSymbol = terminalSymbol;
        }

        public Symbol(NonTerminalSymbol nonTerminalSymbol)
        {
            IsTerminal = false;
            NonTerminalSymbol = nonTerminalSymbol;
        }
    }
}
